server/app.ts

```diff
@@ -581,19 +617,48 @@ apiRouter.route("/checkin").post(authenticateWithReject, postParser, async (requ
 	}
 });
 
+// Endpoint for adding tags to existing groups of attendees
+apiRouter.route("/data/addTag/:tag").put(authenticateWithReject, postParser, async (request, response) => {
+	let tag: string = request.params.tag;
+	let currentTag: string = request.body.currentTag;
+	if (!currentTag) {
+		response.status(400).json({
+			"error": "Must provide existing tag"
+		});
+		return;
+	}
+	let query = {};
+	query["tags." + currentTag] = {$exists: true};
+
+	// Only add tag to attendees that don't already have the tag
+	query["tags." + tag] = {$exists: false};
+	let update: ITags = {};
+	update["tags." + tag] = {checked_in: false};
+	try {
+		await Attendee.update(query, update, {multi: true});
```

Comment:
I don't really understand what this update is doing, it sets `checked_in` to `false` if `currentTag` exists on the user _and_ `tag` does not exist on the user?

Reply:
Also could be written like:
```javascript
await Attendee.update({
	["tags." + currentTag]: {
		$exists: true
	},
	["tags." + tag]: {
		$exists: false
	},
}, {
	["tags." + tag]: {
		checked_in: false
	}
}, {
	multi: true
});
```

Reply:
So I wrote this endpoint for adding tags to groups of attendees, for example let's say we want to add the tag `got-t-shirt` to everyone who already has the tag `hackgt`. In the case that someone who has tag `hackgt` for some reason also already has the tag `got-t-shirt` *and* has already been checked in for `got-t-shirt`, I didn't want to overwrite the fact that they were already checked in for `got-t-shirt`.

Reply:
Also I didn't know it could be written that way. That looks a lot cleaner than the way I did it